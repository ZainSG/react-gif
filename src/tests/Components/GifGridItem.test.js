import {shallow} from 'enzyme'
import GifGridItem from '../../components/GifGridItem'

describe('Pruebas en </GifGridItem>',()=>{
    // tecla u para actualizar el spapshot
    const title = 'A title'
    const url = 'https://localhost/images/image.jpg'
    const wrapper = shallow( <GifGridItem  
        title={title}
        url={url}/>)
    test('should show the comoneny as expected', () => { 
        expect(wrapper).toMatchSnapshot()
    })


    test('Debe de tener un parrafo con el title', () => { 
        const p = wrapper.find('p')
        expect( p.text().trim()).toBe(title)
     })

     test('Debe de obtener la imagen igual a url y a los props', () => { 
         
        const img = wrapper.find('img')
        //console.log(img.html());
       // console.log(img.props().src);
       expect(img.props().src).toBe(url)
       expect(img.props().alt).toBe(title)        
    })

    test('debe tener animate__fadeIn', () => { 
        const div = wrapper.find('div')
        const className = div.props().className
        expect(className.includes('animate__fadeIn')).toBe(true)
     })
})