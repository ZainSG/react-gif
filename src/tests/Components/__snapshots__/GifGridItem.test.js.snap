// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en </GifGridItem> should show the comoneny as expected 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="A title"
    src="https://localhost/images/image.jpg"
  />
  <p>
    A title
  </p>
</div>
`;
